Components/ProductQuery/RemoteProductQuery.php

```diff
@@ -109,6 +124,15 @@ protected function getConnectProduct($row)
         $product = new Product(
             $row
         );
+
+        $this->eventManager->notify(
+            'Connect_Supplier_Get_Single_Product_Before',
```

Comment:
Yes maybe we don't need for products from other suppliers. ProductFromShop means only product where you're supplier. Please remove it from RemoteProductQuery or change event name if we still need to support this.